app/messaging/schemas/sfi-23-quarterly-statement.js

```diff
@@ -37,223 +45,216 @@ const addressSchema = Joi.object({
     'array.base': 'Actions must be an array',
     'array.min': 'Actions array must have at least one item',
-    'any.required': 'Actions array is not present but it is required'
+    'any.required': 'Actions array is required'
   })
 })
 
 module.exports = Joi.object({
   address: addressSchema,
-  businessName: Joi.string().max(number100).required().messages({
+  businessName: Joi.string().max(maxSchemeNameLength).required().messages({
     'string.base': 'Business name must be a string',
-    'string.max': 'Business name must be at most 100 characters',
-    'any.required': 'Business name is not present but it is required'
+    'string.max': `Business name must be at most ${maxSchemeNameLength} characters`,
+    'any.required': 'Business name is required'
   }),
   email: Joi.string().email().required().messages({
     'string.email': 'Email must be a valid email address',
-    'any.required': 'Email is not present but it is required'
+    'any.required': 'Email is required'
   }),
   frn: Joi.number().integer().min(minFrn).max(maxFrn).required().messages({
     'number.base': 'FRN must be a number',
     'number.integer': 'FRN must be an integer',
     'number.min': `FRN must be at least ${minFrn}`,
     'number.max': `FRN must be at most ${maxFrn}`,
-    'any.required': 'FRN is not present but it is required'
+    'any.required': 'FRN is required'
   }),
   sbi: Joi.number().integer().min(minSbi).max(maxSbi).required().messages({
     'number.base': 'SBI must be a number',
     'number.integer': 'SBI must be an integer',
     'number.min': `SBI must be at least ${minSbi}`,
     'number.max': `SBI must be at most ${maxSbi}`,
-    'any.required': 'SBI is not present but it is required'
+    'any.required': 'SBI is required'
   }),
-  paymentReference: Joi.string().max(number30).required().messages({
+  paymentReference: Joi.string().max(maxPaymentReferenceLength).required().messages({
     'string.base': 'Payment reference must be a string',
-    'string.max': 'Payment reference must be at most 30 characters',
-    'any.required': 'Payment reference is not present but it is required'
+    'string.max': `Payment reference must be at most ${maxPaymentReferenceLength} characters`,
+    'any.required': 'Payment reference is required'
   }),
   calculationId: Joi.number().integer().required().messages({
     'number.base': 'Calculation ID must be a number',
     'number.integer': 'Calculation ID must be an integer',
-    'any.required': 'Calculation ID is not present but it is required'
+    'any.required': 'Calculation ID is required'
   }),
-  paymentPeriod: Joi.string().max(number200).allow('', null).optional().messages({
+  paymentPeriod: Joi.string().max(maxPaymentPeriodLength).required().messages({
     'string.base': 'Payment period must be a string',
-    'string.max': 'Payment period must be at most 200 characters'
-  }),
-  paymentAmount: Joi.number().required().messages({
```

Comment:
I could be missing some info but is there a reason we've now removed paymentAmount and transactionDate?

Reply:
omg, I am so done with copilot removing fields for no reason, it's bloody annoying

Reply:
at least we won't be fully replaced by AI for a couple of years...